docs/source/inline_config.rst

```diff
@@ -0,0 +1,40 @@
+.. _inline-config:
+
+Inline configuration
+====================
+
+Mypy supports setting per-file configuration options inside files themselves
+using ``# mypy:`` comments. For example:
+
+.. code-block:: python
+
+  # mypy: disallow-any-generics
+
+Inline configuration comments take precedence over all other
+configuration mechanisms.
+
+Configuration comment format
+****************************
+
+Flags correspond to :ref:`config file flags <config-file>` but allow
+hyphens to be substituted for underscores.
+
+Values are specified using ``=``, but the ``=`` may be omitted to
+specify ``True`` as the value.
+
+Multiple flags can be separated by commas or placed on separate
+lines. To include a comma as part of an option's value, place the
+value inside quotes.
+
+Like in the configuration file, options that take a boolean value may be
+inverted by adding ``no-`` to their name or by (when applicable)
+swapping their prefix from ``disallow`` to ``allow`` (and vice versa).
+
+
+Examples
```

Comment:
I'd prefer to show the relevant example to each feature described, e.g. omitting `=`, multiple flags, `no-` and `dis`, rather than a single example block that shows all possibilities.